src/services/catalog.service.ts

```diff
@@ -21,23 +21,30 @@ export class CatalogService {
     this.logger = Logger;
   }
 
-  public getCatalogItems() {
-    return this.$q.all({
+  public getCatalogItems(includeTemplates: boolean) {
+    let promises: any = {
       serviceClasses: this.dataService.list({
         group: 'servicecatalog.k8s.io',
         resource: 'serviceclasses'
       }, {}),
-      imageStreams: this.dataService.list("imagestreams", {namespace: "openshift"})
-    }).then((values) => {
+      imageStreams: this.dataService.list("imagestreams", { namespace: "openshift" })
+    };
+
+    if (includeTemplates) {
+      promises.templates = this.dataService.list("templates", { namespace: "openshift" });
+    }
+
+    return this.$q.all(promises).then((values) => {
       let serviceClasses: any = values.serviceClasses.by("metadata.name");
       let imageStreams: any = values.imageStreams.by("metadata.name");
-      return this.convertToServiceItems(serviceClasses, imageStreams);
+      let templates: any = values.templates ? values.templates.by("metadata.name") : {};
+      return this.convertToServiceItems(serviceClasses, imageStreams, templates);
     }, () => {
       this.logger.log("Error Loading Catalog Items");
```

Comment:
I'm OK with that, but I think we need to break this into separate methods. It also feels wrong to return a partial result with no indication that something failed.